File: library/src/main/java/com/afollestad/aesthetic/TabLayoutBgMode.java
package com.afollestad.aesthetic;

/** @author Aidan Follestad (afollestad) */
public enum TabLayoutBgMode {
  PRIMARY(0),
  ACCENT(1);

  private final int value;

  TabLayoutBgMode(int value) {
    this.value = value;
  }

  public int toInt() {
    return value;
  }

  public static TabLayoutBgMode fromInt(int value) {
    switch (value) {
      case 0:
        return PRIMARY;
      default:
        return ACCENT;
    }
  }
}

File: library/src/main/java/com/afollestad/aesthetic/TabLayoutBgMode.kt
package com.afollestad.aesthetic

/** @author Aidan Follestad (afollestad)*/
enum class TabLayoutBgMode(val value: Int) {
  PRIMARY(0),
  ACCENT(1);

  companion object {
    fun fromInt(value: Int): TabLayoutBgMode {
      return when (value) {
        0 -> PRIMARY
        else -> ACCENT
      }
    }
  }
}
